### PR TITLE
Added new control

### PR DESCRIPTION
**Pull request by wiotq**

### CODE REVIEW

File: zgui.cc [hunk 1]
```diff
@@ -589,6 +589,72 @@ void zgui::multi_combobox(std::string id, std::vector<multi_item> items)
 	}
 }
 
+void zgui::input_text(std::string id, std::string* value)
```
Comment (cyanidee) [inline zgui.cc]: What about `text_input`?

File: zgui.cc [hunk 2]
```diff
@@ -589,6 +589,72 @@ void zgui::multi_combobox(std::string id, std::vector<multi_item> items)
...
+{
+	std::vector<std::string> id_split = split_str(id, '#');
+
+	const int control_width = 70;
+	const int control_height = 20;
+
+
```
Comment (cyanidee) [inline zgui.cc]: Clean up these empty lines...

File: zgui.cc [hunk 3]
```diff
@@ -589,6 +589,72 @@ void zgui::multi_combobox(std::string id, std::vector<multi_item> items)
...
+						context.window.blocking = 0;
+					break;
+				case VK_ESCAPE:
+					context.window.blocking = 0;
+					break;
+				case VK_SPACE:
+					*value += " ";
+					break;
+				}
+				if(wide < control_width - 18)
```
Comment (cyanidee) [inline zgui.cc]: You shouldn't measure maximum text amount by its width, instead pass another parameter `max_length` or something to the `input_text` method.

File: zgui.cc [hunk 4]
```diff
@@ -589,6 +589,72 @@ void zgui::multi_combobox(std::string id, std::vector<multi_item> items)
...
+					if ((i >=48 && i <= 57) || (i >= 65 && i <= 90))
```
Comment (cyanidee) [inline zgui.cc]: What about [`std::isprint`](http://www.cplusplus.com/reference/cctype/isprint/)?

File: zgui.hh [hunk 5]
```diff
@@ -55,6 +55,9 @@
  *  text(std::string text);
  *  -- function that creates text.
  *
+ * 	input_text(std::string id, std::string* value);
+ *	-- function
```
Comment (cyanidee) [inline zgui.hh]: You should document this function...

**Comment by cyanidee** [comment]
Also, you should reformat your code before making a pull request...

File: zgui.cc (hunk 3 above)
Comment (zxvnme) [reply zgui.cc]: I will fix this later.

**Pull request closed without merging**